### PR TITLE
Cleanup logging 1

### PR DESCRIPTION
**Pull request by rht**
Remove prefixlog (totally unused) and go-logging (backend of util/log).
Now there are only util/log, eventlog, logrus left (util/log and eventlog use logrus as backend).
Side-effect: the logs don't have shortfile (because logrus doesn't use it), and the formatting is less flexible than go-logging. The former needs a fork, the latter only needs a custom formatter.

#593

### CODE REVIEW

File: util/log.go [hunk 1]
```diff
@@ -3,23 +3,21 @@ package util
 import (
 	"os"
 
-	logging "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/whyrusleeping/go-logging"
+	"github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/Sirupsen/logrus"
```
Comment (jbenet) [inline util/log.go]: @whyrusleeping had you switched to `go-logging` on purpose?

**Comment by rht** [comment]
"completely API compatible with the standard library logger" but doesn't have stdlib SetFlags is indeed surprising. https://github.com/Sirupsen/logrus/issues/63

File: util/log.go (hunk 1 above)
Comment (whyrusleeping) [inline util/log.go]: i was under the assumption that we were using go-logging.

Comment (jbenet) [inline util/log.go]: @whyrusleeping do you want to _keep_ go-logging, or is logrus ok? (i dont care either way)

Comment (rht) [inline util/log.go]: @jbenet irc logs and past issues say logrus is the preferred backend. Some of the logs are even in pure logrus. It's just that eventlog heavy logging needs to be toned down somehow.

The argument for go-logging is that it is more lightweight than logrus and more compatible with standard logging extra features (e.g. logrus's setflags feature is still pending).

Comment (rht) [inline util/log.go]: keep go-logging is euphemism for kill logrus (as this will be the course of action written in the commit message, not the former)

Comment (whyrusleeping) [inline util/log.go]: @rht at this point, you know more about them to make a better decision on the two. I'd say take the path of least resistance if theyre about the same.

**Comment by whyrusleeping** [comment]
@rht, status update here?

**Comment by jbenet** [comment]
@whyrusleeping said:

> @rht at this point, you know more about them to make a better decision on the two. I'd say take the path of least resistance if theyre about the same.

so let's go with whatever you think is best, @rht. is that logrus + the changes here?

**Comment by rht** [comment]
@whyrusleeping There is a very specific error that happens here when I switch Critical{,f} with Fatalf{,f}. I'm bisecting on my previous commit (now that gpe is not running). The err is erring.

Yes it is logrus + changes.

**Comment by jbenet** [comment]
@rht is the problem these:

```
17:36:47.598 CRITI   boguskey: TestBogusPrivateKey.Sign -- this better be a test! key.go:54
FAIL	github.com/ipfs/go-ipfs/test/integration	11.337s
```
??

it's because in logrus, `log.Fatalf` calls `os.Exit(1)` after logging: 

> ```go
> // Calls os.Exit(1) after logging
> log.Fatal("Bye.")
> // Calls panic() after logging
> log.Panic("I'm bailing.")
> ```
https://github.com/Sirupsen/logrus#level-logging

Our `Critical` does not cause that sort of failure, so we can probably just change all occurrences of `log.Critical{,f}` to `log.Error{,f}`

**Comment by rht** [comment]
(I c, I did try changing to error, but not to all files...)

**Comment by rht** [comment]
Test passes on the commit "Replace Critical with Error". The latest fail is only on the linux.

**Comment by rht** [comment]
Will close #592, #1217. Cleanup logging 2 will be for #593.

File: exchange/bitswap/wantmanager.go [hunk 2]
```diff
@@ -96,7 +96,7 @@ func (pm *WantManager) SendBlock(ctx context.Context, env *engine.Envelope) {
 	log.Infof("Sending block %s to %s", env.Peer, env.Block)
 	err := pm.network.SendMessage(ctx, env.Peer, msg)
 	if err != nil {
-		log.Noticef("sendblock error: %s", err)
+		log.Warningf("sendblock error: %s", err)
```
Comment (jbenet) [inline exchange/bitswap/wantmanager.go]: this should probably just be `log.Infof` -- warnings are things the user may need to be conscious about

File: exchange/bitswap/wantmanager.go [hunk 3]
```diff
@@ -153,7 +153,7 @@ func (mq *msgQueue) doWork(ctx context.Context) {
 
 	err := mq.network.ConnectTo(conctx, mq.p)
 	if err != nil {
-		log.Noticef("cant connect to peer %s: %s", mq.p, err)
+		log.Warningf("cant connect to peer %s: %s", mq.p, err)
```
Comment (jbenet) [inline exchange/bitswap/wantmanager.go]: this should probably just be `log.Infof` -- warnings are things the user may need to be conscious about

File: exchange/bitswap/wantmanager.go [hunk 4]
```diff
@@ -174,7 +174,7 @@ func (mq *msgQueue) doWork(ctx context.Context) {
 	// send wantlist updates
 	err = mq.network.SendMessage(sendctx, mq.p, wlm)
 	if err != nil {
-		log.Noticef("bitswap send error: %s", err)
+		log.Warningf("bitswap send error: %s", err)
```
Comment (jbenet) [inline exchange/bitswap/wantmanager.go]: this should probably just be `log.Infof` -- warnings are things the user may need to be conscious about

File: ipnsfs/system.go [hunk 5]
```diff
@@ -80,7 +80,7 @@ func (fs *Filesystem) Close() error {
 			defer wg.Done()
 			err := r.Publish(context.TODO())
 			if err != nil {
-				log.Notice(err)
+				log.Warning(err)
```
Comment (jbenet) [inline ipnsfs/system.go]: this should probably just be `log.Info`

**Comment by jbenet** [comment]
last couple changes. otherwise, this LGTM!

**Comment by jbenet** [comment]
@rht, make sure to rebase on latest master. we might merge @whyrusleeping's #1382 if it gets finished first.

**Comment by rht** [comment]
Fixed + rebased

**Comment by jbenet** [comment]
@rht this LGTM! and :+1: on all the go linting.

**Comment by jbenet** [comment]
thanks @rht!

**Comment by rht** [comment]
The golint was an excuse to rerun the travis.

**Pull request merged**